packages/codegen-ui-react/lib/workflow/action.ts

```diff
@@ -122,6 +157,111 @@ export function buildUseActionStatement(
   );
 }
 
+export function buildGraphqlCallback(
+  componentMetadata: ComponentMetadata,
+  action: DataAction,
+  identifier: string,
+  importCollection: ImportCollection,
+) {
+  // Import API client
+  const actionMutation = getGraphqlMutationForModel(action.action as Action, action.parameters.model);
+  importCollection.addMappedImport(ImportValue.API);
+  importCollection.addGraphqlMutationImport(actionMutation);
+  importCollection.addModelImport(action.parameters.model);
+
+  const inputKeys = [];
+
+  if ('id' in action.parameters && action.parameters.id) {
```

Comment:
Should probably null check `action.parameters` as well unless we have validation upstream on the schema.

Reply:
By this point the action has been narrowed to types that have a required parameters property.